code/game/objects/items/shuttle_upgrades.dm

```diff
@@ -0,0 +1,52 @@
+/obj/item/shuttle_route_optimisation
+	name = "Route Optimisation Upgrade"
+	desc = "Used on a shuttle control console to calculate more efficient routes."
+	icon = 'icons/obj/module.dmi'
+	icon_state = "shuttledisk"
+	force = 0
+	throwforce = 8
+	throw_speed = 3
+	throw_range = 5
+	density = FALSE
+	anchored = FALSE
+	item_flags = NOBLUDGEON
+	var/upgrade_amount = 0.8
+
+/obj/item/shuttle_route_optimisation/hyperlane
+	name = "Bluespace Hyperlane Calculator"
+	desc = "Used on a shuttle control console to allow for the following of bluespace hyperlanes, increasing the efficiency of the shuttle."
+	icon = 'icons/obj/module.dmi'
+	icon_state = "shuttledisk_better"
+	force = 0
+	throwforce = 8
+	throw_speed = 3
+	throw_range = 5
+	density = FALSE
+	anchored = FALSE
+	item_flags = NOBLUDGEON
+	upgrade_amount = 0.6
+
+/obj/item/shuttle_route_optimisation/void
+	name = "Voidspace Route Calculator"
+	desc = "Used on a shuttle control console to allow it to navigate into voidspace, making the routes almost instant."
+	icon = 'icons/obj/module.dmi'
+	icon_state = "shuttledisk_void"
+	force = 0
+	throwforce = 8
+	throw_speed = 3
+	throw_range = 5
+	density = FALSE
+	anchored = FALSE
+	item_flags = NOBLUDGEON
+	upgrade_amount = 0.2
+
+/obj/item/shuttle_route_optimisation/attack_obj(obj/O, mob/living/user)
+	. = ..()
+	if(!istype(O, /obj/machinery/computer/custom_shuttle))
+		return
```

Comment:
Should probably provide feedback to the user that it needs to be used on a shuttle control console. Also it should be clear that this is for custom shuttles, not existing shuttles.